synapse/util/ratelimitutils.py

```diff
@@ -59,7 +66,7 @@ def ratelimit(self, host: str) -> "_GeneratorContextManager[defer.Deferred[None]
         Returns:
             context manager which returns a deferred.
         """
-        return self.ratelimiters[host].ratelimit()
+        return self.ratelimiters[host].ratelimit(host)
```

Comment:
Better way to pass the string `host` to the `_PerHostRateLimiter`?

There doesn't seem to be a good way to make `DefaultDict` pass the string key host on, https://stackoverflow.com/questions/2912231/is-there-a-clever-way-to-pass-the-key-to-defaultdicts-default-factory